### PR TITLE
Implements iterator interface for vector functions

### PR DESCRIPTION
**Pull request by blegat**

### CODE REVIEW

**Comment by mlubin** [comment]
I'm not sure I see vector functions naturally as collections of scalar functions. What about making this iteration interface accessible through `for scalar_function in eachscalar(f)` instead of `for scalar_function in f`?

**Comment by blegat** [comment]
So we implement indexing directly (so `getindex`, `length` and `endof`) and iteration through `eachscalar` ?

**Comment by mlubin** [comment]
I'd say that makes sense, definitely clearer.

File: src/functions.jl [hunk 1]
```diff
@@ -6,11 +6,37 @@ function MOI.VectorAffineFunction{T}(f::MOI.VectorOfVariables) where T
...
+Base.eltype(f::MOI.VectorOfVariables) = MOI.SingleVariable
+Base.eltype(f::MOI.VectorAffineFunction{T}) where T = MOI.ScalarAffineFunction{T}
+Base.eltype(f::MOI.VectorQuadraticFunction{T}) where T = MOI.ScalarQuadraticFunction{T}
+
+Base.getindex(f::MOI.VectorOfVariables, i::Integer) = MOI.SingleVariable(f.variables[i])
 function Base.getindex(f::MOI.VectorAffineFunction, i::Integer)
     I = find(oi -> oi == i, f.outputindex)
     MOI.ScalarAffineFunction(f.variables[I], f.coefficients[I], f.constant[i])
 end
+function Base.getindex(f::MOI.VectorQuadraticFunction, i::Integer)
```
Comment (mlubin) [inline src/functions.jl]: Shouldn't `getindex` apply to `ScalarFunctionIterator`, not the MOI function directly?

Comment (blegat) [reply src/functions.jl]: Ah ok, I misunderstood. That is changed now.

**Review by mlubin — APPROVED** [review]

**Pull request merged**